packages/docusaurus-theme-classic/src/theme/Tabs/index.tsx

```diff
@@ -69,16 +79,15 @@ function TabsComponent(props: Props): JSX.Element {
         .join(', ')}" found in <Tabs>. Every value needs to be unique.`,
     );
   }
-  // When defaultValueProp is null, don't show a default tab
-  const defaultValue =
-    defaultValueProp === null
-      ? defaultValueProp
-      : defaultValueProp ??
-        children.find((child) => child.props.default)?.props.value ??
-        children[0]!.props.value;
-  if (defaultValue !== null && !values.some((a) => a.value === defaultValue)) {
+
+  // Warn user about passing incorrect defaultValue as prop.
+  if (
+    defaultValueProp !== null &&
+    defaultValueProp !== undefined &&
+    !values.some((a) => a.value === defaultValueProp)
+  ) {
```

Comment:
can you explain the intent here? Not sure to understand

Reply:
Very similar to the [prev check](https://github.com/facebook/docusaurus/blob/e41133245ca4e6cee390da74ccfe305ccb2ddb25/packages/docusaurus-theme-classic/src/theme/Tabs/index.tsx#L79). However, this new one only throws in case of user-error. E.g. available tab keys are `ke1`, `key2`, `key3`, but user supplies `key4`. The goal is to notify user it's their fault that the tabs are not working properly.

The prev implementation would throw even if the error was on the Docusaurus (implementation) side as it checked the derived value instead of user supplied one.